mysql-test/suite/innodb/r/innodb-fk-warnings.result

```diff
@@ -16,7 +16,7 @@ CONSTRAINT test FOREIGN KEY (b) REFERENCES t2 (id)
 ERROR HY000: Can't create table `test`.`t2` (errno: 121 "Duplicate key on write or update")
 show warnings;
 Level	Code	Message
-Warning	121	Create or Alter table `test`.`t2` with foreign key constraint failed. Foreign key constraint `test`.`test` already exists on data dictionary. Foreign key constraint names need to be unique in database. Error in foreign key definition: CONSTRAINT `test` FOREIGN KEY (`b`) REFERENCES `test`.`t2` (`id`).
+Warning	121	CREATE or ALTER TABLE `test`.`t2` failed, CONSTRAINT `test` FOREIGN KEY (`b`) REFERENCES `t2` (`id`)
```

Comment:
Looks like the precision of the error messages degraded. The reason for this failure is existing constraint named "test" and the current error message at least explains this part.

> Foreign key constraint `test`.`test` already exists on data dictionary. Foreign key constraint names need to be unique in database.

It usually is annoying when an error is thrown and I cannot make out what is the problem. Perhaps along with the general error/warning we can preserve the important information for each specific code path by pushing mote warning to the stack.

Reply:
I thought that the information was redundant, because also the following errors would be emitted for the same statement:
```
Error	1005	Can't create table `test`.`t2` (errno: 121 "Duplicate key on write or update")
Warning	1022	Can't write; duplicate key in table 't2'
```
But, I now realize that it can be a little confusing. I think that we can improve this by adding `: duplicate name` right before the dump that starts with `, CONSTRAINT`.